test/tools/runner/config.js

```diff
@@ -104,13 +109,13 @@ class TestConfiguration {
   }
 
   newClient(dbOptions, serverOptions) {
-    const defaultOptions = { minHeartbeatFrequencyMS: 100 };
-    if (this.serverApi) {
-      Object.assign(defaultOptions, { serverApi: this.serverApi });
-    }
+    const defaultOptions = Object.assign(
+      { minHeartbeatFrequencyMS: 100 },
+      getEnvironmentalOptions()
+    );
     // support MongoClient constructor form (url, options) for `newClient`
     if (typeof dbOptions === 'string') {
-      return new MongoClient(dbOptions, Object.assign(defaultOptions, serverOptions));
+      return new MongoClient(dbOptions, Object.assign({}, defaultOptions, serverOptions));
```

Comment:
since the defaultOptions are generated in this test, why generate a new object instead of just passing in the updated defaultOptions as before?

Reply:
Good point, I'll clean this up 👍

Reply:
Refactored, PTAL 🙏